pkg/apis/core/types_shoot.go

```diff
@@ -984,6 +984,8 @@ type Worker struct {
 	Zones []string
 	// MachineControllerManagerSettings contains configurations for different worker-pools. Eg. MachineDrainTimeout, MachineHealthTimeout.
 	MachineControllerManagerSettings *MachineControllerManagerSettings
+	// Architecture is CPU architecture of machines in this worker pool.
+	Architecture *string
```

Comment:
Wouldn't it be better to place this field under the machine field?

```yaml
    workers:
    - name: cpu-worker
      minimum: 3
      maximum: 5
      machine:
        type: m5.large
        architecture: amd64
        image:
          name: <some-image-name>
          version: <some-image-version>
```

cc @rfranzke